File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int i=0;
        int j=0;
        int max_win=-1;
        while(j<nums.length)
        {
            if(nums[j]==0)
            {
                k--;
            }
            while(k==-1)
            {
                if(nums[i]==0)
                {
                    k++;
                }
                i++;
            }
            int curr_win=j-i+1;
            max_win=Math.max(max_win,curr_win);
            j++;
        }
        return max_win;
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        i=0
        j=0
        max_win=-1
        while(j<len(nums)):
            if(nums[j]==0):
                k-=1
            while(k==-1):
                if(nums[i]==0):
                    k+=1
                i+=1
            curr_win=j-i+1
            max_win=max(max_win,curr_win)
            j+=1
        return max_win